src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -461,7 +461,7 @@ namespace cryptonote
 
       // the non-simple version is slightly smaller, but assumes all real inputs
       // are on the same index, so can only be used if there just one ring.
-      bool use_simple_rct = sources.size() > 1;
+      bool use_simple_rct = sources.size() > 1 || bulletproof;
```

Comment:
Why is the Simple RingCT always used when using BP?

Reply:
Because it was only done for simple originally. I will remove this now.